### PR TITLE
[BOOST-4684] test: add agora vote example

### PR DESCRIPTION
**Pull request by jonathandiep**
### Description
Adds the agora vote example as part of the viability tests. It shows how people can create a Boost to incentivize voting for/against on certain proposals

Some features in this test:
- Multiple action steps, each targeting different params in the same event
- Dynamic rewards (via `ERC20VariableIncentive`)

### CODE REVIEW

File: examples/agora-vote/hardhat.config.cjs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+require('dotenv').config();
+const { optimism } = require('viem/chains');
```
Comment (jonathandiep) [inline examples/agora-vote/hardhat.config.cjs:2]: Is there any issue with using public rpc for this? lol

Comment (Quazia) [reply examples/agora-vote/hardhat.config.cjs:2]: Only issue I can see is potentially blocking CI but not a huge deal 🤷

File: examples/agora-vote/src/agora-vote.test.ts [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+          ],
+          stateMutability: 'view',
+          type: 'function',
+        },
+      ],
+      functionName: 'getVotes',
+      args: [boostImpostor, txReceipt.blockNumber],
+    });
+
+    console.log({ amountOfVotes})
```
Comment (mmackz) [inline examples/agora-vote/src/agora-vote.test.ts:233]: clean up log 🪵

**Comment by mmackz** [comment]
I think we need a changeset for the signatures packages whenever we add a new signature.

File: examples/agora-vote/src/agora-vote.test.ts [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+    const { core } = fixtures;
+
+    const owner = defaultOptions.account.address;
+
+    // Step defining the action for VoteCast event
+    const eventActionStepOne: ActionStep = {
+      chainid: optimism.id,
+      signature: selector, // VoteCast event signature
+      signatureType: SignatureType.EVENT, // We're working with an event
+      actionType: 0, // Custom action type (set as 0 for now)
```
Comment (sammccord) [inline examples/agora-vote/src/agora-vote.test.ts:89]: you can omit `actionType`

Comment (sammccord) [inline examples/agora-vote/src/agora-vote.test.ts:89]:
```suggestion
```

Comment (sammccord) [reply examples/agora-vote/src/agora-vote.test.ts:89]: actionType defaults to 0

File: examples/agora-vote/src/agora-vote.test.ts [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+          ),
+        ), // Filtering based on the proposal id
+      },
+    };
+
+    const eventActionStepTwo: ActionStep = {
+      chainid: optimism.id,
+      signature: selector, // VoteCast event signature
+      signatureType: SignatureType.EVENT, // We're working with an event
+      actionType: 0, // Custom action type (set as 0 for now)
```
Comment (sammccord) [inline examples/agora-vote/src/agora-vote.test.ts:108]:
```suggestion
```

File: examples/agora-vote/src/agora-vote.test.ts [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+  });
+
+  test('should create a boost for incentivizing votes', async () => {
+    const { budget, erc20 } = budgets;
+    const { core } = fixtures;
+
+    const owner = defaultOptions.account.address;
+
+    // Step defining the action for VoteCast event
+    const eventActionStepOne: ActionStep = {
```
Comment (sammccord) [inline examples/agora-vote/src/agora-vote.test.ts:85]: let's name these steps something more semantic, like `filterProposalIdStep`

File: examples/agora-vote/src/agora-vote.test.ts [hunk 6]
```diff
@@ -0,0 +1,263 @@
...
+      toBlock: 'latest',
+    });
+    const validation = await action.validateActionSteps({
+      logs,
+    });
+    expect(validation).toBe(true);
+
+    const amountOfVotes = await walletClient.readContract({
+      address: '0xcdf27f107725988f2261ce2256bdfcde8b382b10',
+      abi: [
```
Comment (sammccord) [inline examples/agora-vote/src/agora-vote.test.ts:203]: if we add the getVotes function to packages/signatures/manifests/functions.json, we could pull this AbiItem off `functions.abi[functions.selectors['getVotes(...)']]`, maybe we can also index the AbiItems with the name as well to make that simpler

**Review by sammccord — APPROVED** [review]
mostly nits but 👍🏼

**Pull request merged**